### PR TITLE
utils file for common functions and constants

### PR DESCRIPTION
**Pull request by nishaq503**
feat: broke out utils file for common usage

### CODE REVIEW

File: pyclam/chaoda.py [hunk 1]
```diff
@@ -214,21 +188,21 @@ def random_walk(graph: Graph, normalize: str = 'gaussian') -> np.array:
 
     # normalize counts
     scores: Dict[int, int] = {point: -count for cluster, count in visit_counts.items() for point in cluster.argpoints}
-    return _normalize(np.array([scores[i] for i in range(graph.population)], dtype=int), mode=normalize)
+    return normalize(np.array([scores[i] for i in range(graph.population)], dtype=int), mode=normalization_mode)
 
 
-def stationary_probabilities(graph: Graph, normalize: str = 'gaussian') -> np.array:
+def stationary_probabilities(graph: Graph, normalization_mode: str = 'gaussian') -> np.array:
```
Comment (thoward27) [inline pyclam/chaoda.py:194]: I think we should use this: https://docs.python.org/3/library/typing.html#typing.Literal

File: pyclam/tests/utils.py [hunk 2]
```diff
@@ -15,24 +15,3 @@ def linear_search(point: Data, radius: Radius, data: Data, metric: str):
         distances = cdist(point, batch, metric)[0]
         results.extend([p for p, d in zip(batch, distances) if d <= radius])
     return results
-
-
-def trace_lineage(cluster: Cluster, other: Cluster):  # TODO: Cover
```
Comment (thoward27) [inline pyclam/tests/utils.py:20]: Is this still functional? Do we see a potential need for it in the future?

File: pyclam/utils.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+def catch_normalization_mode(mode: str) -> None:
+    modes: List[str] = ['linear', 'gaussian', 'sigmoid']
+    if mode not in modes:
+        raise ValueError(f'Normalization method {mode} is undefined. Must by one of {modes}.')
+    else:
+        return
+
+
+def normalize(scores: np.array, mode: str) -> np.array:
```
Comment (thoward27) [inline pyclam/utils.py:18]: docstring

File: pyclam/utils.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
+from typing import List
+
+import numpy as np
+from scipy.special import erf
+
+SUBSAMPLE_LIMIT = 100
+BATCH_SIZE = 10_000
+
+
+def catch_normalization_mode(mode: str) -> None:
```
Comment (thoward27) [inline pyclam/utils.py:10]: docstring

File: pyclam/utils.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
+from typing import List
```
Comment (thoward27) [inline pyclam/utils.py:1]: maybe add module docstring

File: pyclam/utils.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+import numpy as np
+from scipy.special import erf
+
+SUBSAMPLE_LIMIT = 100
+BATCH_SIZE = 10_000
```
Comment (thoward27) [inline pyclam/utils.py:7]: I don't see a direct use of these in this file. Perhaps we should consider a `constants.py` file

**Review by thoward27 — CHANGES_REQUESTED** [review]
requesting changes for docstrings, also had a typing nit and a suggestion around constants

File: pyclam/chaoda.py (hunk 1 above)
Comment (nishaq503) [reply pyclam/chaoda.py:194]: I think we should have a separate PR for using literals. I think we should treat this more like we treat Metric. some strings to represent build-in functionality, while also allowing a user-defined function. In either case, I think that's outside the scope of this PR

File: pyclam/tests/utils.py (hunk 2 above)
Comment (nishaq503) [reply pyclam/tests/utils.py:20]: It's not used anywhere and I don't see it being used anywhere in the future. If I'm wrong, it's still in the git history

File: pyclam/utils.py (hunk 6 above)
Comment (nishaq503) [reply pyclam/utils.py:9]: We can talk about a constants.py or a config.py file but I think that such a thing should be its own PR

**Review by nishaq503 — COMMENTED** [review]
I think that using the Literal type for normalization modes, and maybe also fro Metric, is a good idea. Also, breaking out constants and some other things, like setting up logging, into a config.py file will be a good idea.

However, I think that those changes should be their own PR and are ouside the scope of this PR.

**Review by thoward27 — APPROVED** [review]
LGTM

**Pull request merged**